controllers/tests/controller_server/host_definer/utils/k8s_manifests_utils.py

```diff
@@ -5,93 +5,110 @@
 
 
 def get_k8s_csi_node_manifest(csi_provisioner):
-    return {
-        settings.METADATA: {
-            settings.NAME: settings.FAKE_NODE_NAME
-        },
+    metadata_manifest = _generate_metadata_manifest(settings.FAKE_NODE_NAME)
+    k8s_csi_node_spec = {
         settings.SPEC: {
             settings.DRIVERS: [{
                 settings.NAME: csi_provisioner,
                 settings.NODE_ID_FIELD_IN_CSI_NODE: settings.FAKE_NODE_ID
             }]
         },
     }
+    return _merge_dicts(metadata_manifest, k8s_csi_node_spec)
 
 
 def get_fake_k8s_daemon_set_manifest(updated_pods, desired_updated_pods):
-    return {
-        settings.METADATA: {
-            settings.NAME: settings.FAKE_DAEMON_SET_NAME
-        },
+    metadata_manifest = _generate_metadata_manifest(settings.FAKE_NODE_PODS_NAME)
+    k8s_daemon_set_status = {
         settings.STATUS: {
             settings.UPDATED_PODS: updated_pods,
             settings.DESIRED_UPDATED_PODS: desired_updated_pods,
-        }
-    }
+        }}
+    return _merge_dicts(metadata_manifest, k8s_daemon_set_status)
```

Comment:
you can send "extra dict" to `_generate_metadata_manifest` and save all the "merge" duplication